src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -8852,6 +8910,703 @@ class B<T>
+@"(1, System.Int32, System.String)
+<>f__AnonymousDelegate0`2[System.Int32,System.String]
+(1, System.String, System.Int32)
+<>f__AnonymousDelegate0`2[System.String,System.Int32]
+(2, System.Int32, System.String)
+<>f__AnonymousDelegate0`2[System.Int32,System.String]
+(2, System.String, System.Int32)
+<>f__AnonymousDelegate0`2[System.String,System.Int32]
+(3, System.String, System.Int32)
+<>f__AnonymousDelegate1`2[System.Int32,System.String]
+(3, System.Int32, System.String)
+<>f__AnonymousDelegate1`2[System.String,System.Int32]
+");
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_34()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void Main()
+    {
+        Report(A<int>.B<string>.M1<object>());
+        Report(A<int>.M2());
+        Report(A<int>.M3<string>());
+        Report(A<int>.M4<string, object>());
+        Report(A<string>.B<object>.M1<int>());
+        Report(A<string>.M2());
+        Report(A<string>.M3<object>());
+        Report(A<string>.M4<object, int>());
+    }
+    static void Report(Delegate d) => Console.WriteLine(d.GetType());
+}
+struct A<T>
+{
+    internal class B<U>
+    {
+        internal unsafe static Delegate M1<V>()
+        {
+            var d = void (int* x) => { Console.WriteLine(((int)x, typeof(T), typeof(U), typeof(V))); };
+            d.Invoke((int*)1);
+            return d;
+        }
+    }
+    internal unsafe static Delegate M2()
+    {
+        var d = void (int* x) => { Console.WriteLine(((int)x, typeof(T))); };
+        d.Invoke((int*)2);
+        return d;
+    }
+    internal unsafe static Delegate M3<U>()
+    {
+        var d = void (int* x) => { Console.WriteLine(((int)x, typeof(T), typeof(U))); };
+        d.Invoke((int*)3);
+        return d;
+    }
+    internal unsafe static Delegate M4<U, V>()
+    {
+        var d = void (int* x) => { Console.WriteLine(((int)x, typeof(T), typeof(U), typeof(V))); };
+        d.Invoke((int*)4);
+        return d;
+    }
+}";
+            CompileAndVerify(source, options: TestOptions.UnsafeReleaseExe, verify: Verification.Skipped, expectedOutput:
+@"(1, System.Int32, System.String, System.Object)
+<>f__AnonymousDelegate0
+(2, System.Int32)
+<>f__AnonymousDelegate0
+(3, System.Int32, System.String)
+<>f__AnonymousDelegate0
+(4, System.Int32, System.String, System.Object)
+<>f__AnonymousDelegate0
+(1, System.String, System.Object, System.Int32)
+<>f__AnonymousDelegate0
+(2, System.String)
+<>f__AnonymousDelegate0
+(3, System.String, System.Object)
+<>f__AnonymousDelegate0
+(4, System.String, System.Object, System.Int32)
+<>f__AnonymousDelegate0
+");
+        }
+
+        [Fact]
+        public void SynthesizedDelegateTypes_35()
+        {
+            var sourceA =
+@".class public A`1<T>
+{
+  .method public hidebysig specialname rtspecialname instance void .ctor() cil managed { ret }
+  .method public static void F1(int32* modopt(class A`1<!T>) i) { ret }
```

Comment:
Consider expanding this test to observe the modopt in the synthesized delegate type

Reply:
The `modopt` is used to test that the resulting delegate is generic only (the type parameter is referenced in the `modopt` only), but the resulting delegate does not include a `modopt`.